From f84deddb101230db09c74ff7c750f795c46d0239 Mon Sep 17 00:00:00 2001
From: "an.xi" <an.xi@amlogic.com>
Date: Fri, 23 Jul 2021 09:59:22 +0800
Subject: [PATCH] cec: no more show volume bar if passthrough condition is not
 ok [1/1]

PD#SWPL-56084

Problem:
Volume bar still shows and is covered by warning toast when box
can't process dolby audio format locally.

Solution:
As google suggests and Toast widgest can't be customized, then just
no more show volume bar if passthrough condition is not ok.

Verify:
t3 on p and r.

Change-Id: I9814fdfd58b21d5fc31fde51a05a803a5a59a88a
Signed-off-by: an.xi <an.xi@amlogic.com>
---
 .../com/android/server/audio/AudioService.java  | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/services/core/java/com/android/server/audio/AudioService.java b/services/core/java/com/android/server/audio/AudioService.java
index 2d4cafe62965..db9ad94a2914 100755
--- a/services/core/java/com/android/server/audio/AudioService.java
+++ b/services/core/java/com/android/server/audio/AudioService.java
@@ -2241,7 +2241,9 @@ public class AudioService extends IAudioService.Stub
         int oldIndex = mStreamStates[streamType].getIndex(device);
         if ("WindowManager".equals(caller) || !isMuteAdjust) {
             synchronized (mHdmiClientLock) {
-                passthroughToTv(streamType, direction, oldIndex, oldIndex, keyEventMode);
+                if (passthroughToTv(streamType, direction, oldIndex, oldIndex, keyEventMode)) {
+                    return;
+                }
             }
         }
 
@@ -2765,6 +2767,12 @@ public class AudioService extends IAudioService.Stub
 
             index = rescaleIndex(index * 10, streamType, streamTypeAlias);
 
+            synchronized (mHdmiClientLock) {
+                if (passthroughToTv(streamType, 0, oldIndex, index, VOL_ADJUST_NORMAL)) {
+                    return;
+                }
+            }
+
             if (streamTypeAlias == AudioSystem.STREAM_MUSIC
                     && AudioSystem.DEVICE_OUT_ALL_A2DP_SET.contains(device)
                     && (flags & AudioManager.FLAG_BLUETOOTH_ABS_VOLUME) == 0) {
@@ -2815,7 +2823,6 @@ public class AudioService extends IAudioService.Stub
                     && (oldIndex != index)) {
                 maybeSendSystemAudioStatusCommand(false);
             }
-            passthroughToTv(streamType, 0, oldIndex, index, VOL_ADJUST_NORMAL);
         }
         sendVolumeUpdate(streamType, oldIndex, index, flags, device);
     }
@@ -7466,13 +7473,11 @@ public class AudioService extends IAudioService.Stub
             showPassthroughToast();
             return true;
         }
-        if (DEBUG_VOL) {
-            Slog.d(TAG, "passthroughToTv cec disabled or tv no support cec!");
-        }
+        Slog.d(TAG, "passthroughToTv cec disabled or tv no support cec!");
 
         // show a warning to help the user switch to tv's remote when no key events is sent.
         showPassthroughWarning();
-        return false;
+        return true;
     }
 
     private void showPassthroughToast() {
-- 
2.26.1

